include/circt/Dialect/OM/OMOps.td

```diff
@@ -58,4 +58,25 @@ def ClassFieldOp : OMOp<"class.field",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Object instantiations
+//===----------------------------------------------------------------------===//
+
+def ObjectOp : OMOp<"object",
+    [DeclareOpInterfaceMethods<SymbolUserOpInterface>]> {
```

Comment:
Can we mark this Pure or do you have a plan to allow objects to have side-effects for some sense?

Reply:
Good question, I had thought a little bit about this... and in general I guess maybe we could have a world where this is not pure. But now that you bring it up, that's probably not a world we want to live in :) So I will add Pure, and we should maintain this restriction unless we really need to relax it.